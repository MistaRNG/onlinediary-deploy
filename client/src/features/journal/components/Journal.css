.Journal {
  display: flex;
  width: 100%;
  height: 100%;
  margin: 4em 0 1.25em 0;
  padding-bottom: 1.75em;
  flex-direction: column;
  align-items: center;
  overflow-y: scroll;
}

.bright .Editor,
.bright .Journal input.bp4-input,
.bright .Editor textarea.bp4-input.bp4-large {
  background: #f9f9f9;
}

.dark .Editor,
.dark .Journal input.bp4-input,
.dark .Editor textarea.bp4-input.bp4-large {
  background: #353538;
}

.journal-header {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  padding-right: 20px;
}

.header-buttons {
  display: flex;
  align-items: center;
  margin-left: auto;
}

.entries-button {
  padding: 8px 16px;
  border-radius: 4px;
  background-color: #007bff;
  color: white;
  font-size: 1em;
  text-transform: uppercase;
  transition: background-color 0.3s ease;
  margin-right: 10px;
}

.entries-button:hover {
  background-color: #0056b3;
}

.logout-button-container {
  display: flex;
  align-items: center;
  margin-left: 10px;
}

@media (min-width: 420px) {
  .Journal {
    margin: 6em 0 2.5em 0;
    padding: 0 6em 3.5em 6em;
    align-items: stretch;
  }
}

@media (min-width: 900px) {
  .Journal {
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
  }
}
