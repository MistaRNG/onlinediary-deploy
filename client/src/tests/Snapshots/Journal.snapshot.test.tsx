import React from 'react';
import { render } from '@testing-library/react';
import Journal from '../../features/journal/components';
import '@testing-library/jest-dom';
import { describe, it, expect } from '@jest/globals';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

const mockReducer = (state = { data: [], saved: false }, action: any) => state;

const mockStore = configureStore({
  reducer: {
    journals: mockReducer,
  },
});

describe('Journal Component Snapshot Tests', () => {
  it('should render the Journal component correctly', () => {
    // ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <MemoryRouter initialEntries={['/journal/2024-09-06']}>
          <Journal />
        </MemoryRouter>
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
