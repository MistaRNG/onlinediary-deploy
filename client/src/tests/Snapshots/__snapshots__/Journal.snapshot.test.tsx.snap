// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal Component Snapshot Tests should render the Journal component correctly 1`] = `
<DocumentFragment>
  <style />
  <div
    class="Logout-Button"
  >
    <button
      class="bp4-button"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-people"
        icon="people"
      >
        <svg
          data-icon="people"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M13.69 13.98c-.05-.24-.14-.5-.25-.76-.36-.86-1.12-1.33-2.69-2-.14-.06-.59-.25-.6-.25-.21-.09-.36-.15-.5-.22.02-.1.02-.2.03-.31 0-.04.01-.08.01-.13-.07-.06-.13-.12-.19-.19.22-.32.4-.67.54-1.05.02-.06.02-.06.03-.1.29-.23.48-.57.59-.96.16-.33.25-.73.21-1.16-.03-.4-.16-.76-.37-1.03-.02-.53-.07-1.13-.15-1.54-.01-.06-.02-.12-.03-.19.23-.06.48-.09.72-.09.49 0 1.05.16 1.44.46.38.29.67.7.8 1.17.03.1.05.21.07.31.07.37.11.94.11 1.33v.05c.14.06.27.21.29.51.02.25-.07.45-.13.54-.05.21-.16.44-.38.48-.02.1-.05.2-.09.3 0 .01-.01.03-.01.03-.17.44-.43.83-.75 1.11v.14c.03.35-.09.59.83 1 .93.41 2.32.84 2.6 1.5.29.66.17 1.04.17 1.04h-2.3zm-1.17-.38c.37.86.22 1.36.22 1.36H.06s-.14-.5.22-1.36 2.13-1.43 3.31-1.96c1.17-.54 1.05-.86 1.09-1.3 0-.05.01-.11.01-.17-.41-.35-.75-.86-.97-1.45v-.01s-.01-.01-.01-.02c-.04-.12-.09-.26-.12-.39-.28-.05-.44-.36-.5-.64-.06-.12-.19-.39-.16-.71.04-.41.21-.6.39-.68v-.06c0-.51.05-1.26.14-1.74.02-.13.05-.27.09-.4.17-.6.54-1.13 1.02-1.51.5-.39 1.21-.6 1.84-.6s1.34.21 1.84.6c.48.38.85.91 1.02 1.52.04.13.07.27.09.4.09.48.14 1.22.14 1.73v.07c.18.08.34.27.37.67.03.32-.09.59-.16.71-.06.28-.21.58-.48.63-.03.13-.07.26-.12.39 0 .01-.01.04-.01.04-.22.58-.55 1.08-.95 1.45v.18c.04.45-.12.77 1.06 1.3 1.18.53 2.95 1.09 3.31 1.95z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp4-button-text"
      >
        Entries from others
      </span>
    </button>
    <button
      class="bp4-button"
      type="button"
    >
      <span
        class="bp4-button-text"
      >
        Logged in as @
      </span>
    </button>
    <button
      class="bp4-button"
      type="button"
    >
      <span
        aria-hidden="true"
        class="bp4-icon bp4-icon-log-out"
        icon="log-out"
      >
        <svg
          data-icon="log-out"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M7 14H2V2h5c.55 0 1-.45 1-1s-.45-1-1-1H1C.45 0 0 .45 0 1v14c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zm8.71-6.71l-3-3a1.003 1.003 0 00-1.42 1.42L12.59 7H6c-.55 0-1 .45-1 1s.45 1 1 1h6.59l-1.29 1.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3-3c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    class="bp4-spinner"
    role="progressbar"
  >
    <div
      class="bp4-spinner-animation"
    >
      <svg
        height="50"
        stroke-width="8.00"
        viewBox="1.00 1.00 98.00 98.00"
        width="50"
      >
        <path
          class="bp4-spinner-track"
          d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
        />
        <path
          class="bp4-spinner-head"
          d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          pathLength="280"
          stroke-dasharray="280 280"
          stroke-dashoffset="210"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
